packages/modelence/src/cron/jobs.ts

```diff
@@ -10,25 +11,34 @@ import { Store } from '../data/store';
 const DEFAULT_TIMEOUT = time.minutes(1);
 
 /**
- * Each cron instance acquires locks for the jobs it runs. If there was a pre-existing lock,
- * the lock is transferred to the new instance, but there is a delay to give the previous instance
- * a chance to see the new lock and gracefully finish remaining jobs.
+ * Interval at which lock heartbeats are updated in the database
  */
-const LOCK_TRANSFER_DELAY = time.seconds(10);
+const LOCK_HEARTBEAT_INTERVAL = time.seconds(5);
+
+/**
+ * Time after which a lock is considered stale if no heartbeat update was received
+ * Should be significantly larger than LOCK_HEARTBEAT_INTERVAL to account for network delays
+ */
+const LOCK_STALE_THRESHOLD = time.seconds(30);
 
 const cronJobs: Record<string, CronJob> = {};
 let cronJobsInterval: NodeJS.Timeout;
 
+/**
+ * Unique identifier for this cron instance.
+ * Generated once per application instance to track which container owns which locks.
+ */
+const containerId = randomBytes(32).toString('base64url');
+
 const cronJobsCollection = new Store('_modelenceCronJobs', {
   schema: {
     alias: schema.string(),
     lastStartDate: schema.date().optional(),
-    lock: schema
-      .object({
-        containerId: schema.string(),
-        acquireDate: schema.date(),
-      })
-      .optional(),
+    lock: schema.object({
```

Comment:
Shouldn't this still be `.optional()`? Do we have a lock as soon as the cron job is inserted?